src/Elm/Kernel/Utils.js

```diff
@@ -178,16 +178,6 @@ function _Utils_ap(xs, ys)
 		return xs + ys;
 	}
 
-	// append Lists
-	if (!xs.b)
-	{
-		return ys;
-	}
-	var root = __List_Cons(xs.a, ys);
-	xs = xs.b
-	for (var curr = root; xs.b; xs = xs.b) // WHILE_CONS
-	{
-		curr = curr.b = __List_Cons(xs.a, ys);
-	}
-	return root;
+        // append Lists
+        __List_ap(xs, ys);
```

Comment:
Unless I am missing soemthing  (which I may well be), a return is needed here otherwise this is a nip. Unless this function appends in place? 
```suggestion
        return __List_ap(xs, ys);
```